File: tools/extract-changes.hs
#!/usr/bin/env stack
-- stack --stack-yaml=stack.yaml runghc --package pandoc-types

-- Extract changes from latest version in changelog.
import Text.Pandoc.JSON

main = toJSONFilter extractFirst

extractFirst :: Pandoc -> Pandoc
extractFirst (Pandoc meta bs) =
  let bs' = dropWhile (not . isSubhead) bs
   in Pandoc meta (takeWhile (not . isSubhead) (drop 1 bs'))

isSubhead (Header 2 _ _) = True
isSubhead _ = False

File: tools/extract-changes.lua
-- Extract changes from latest version in changelog.

function Pandoc(el)
  local newblocks = {}
  i = 1
  while i <= #el.blocks and
      not (el.blocks[i].t == "Header" and el.blocks[i].level == 2) do
    i = i+1
  end
  while i <= #el.blocks do
    i = i+1
    if el.blocks[i].t == "Header" and el.blocks[i].level == 2 then
      break
    end
    table.insert(newblocks, el.blocks[i])
  end
  return pandoc.Pandoc(newblocks)
end
